src/core/services/theming/colors.js

```diff
@@ -0,0 +1,137 @@
+(function () {
+    "use strict";
+
+    //module global to available color pallettes
+    var colorStore     = {};
+    var defaultPalette = '';
+
+/**
+ * @ngdoc service
+ * @name $mdColors
+ * @module material.core.color
+ *
+ * @description
+ *  $mdColors makes it possible to use the colors of any palette to be available
+ *  as CSS rules in the form of md-bg-[colorname] and md-fg-[colorname] (as foreground and background)
+ *  If you want to activate another palette you can use the `$mdColor.loadPalette(palletteName)` method.
+ *
+ *  the $mdColors.palettes array lists all available palettes
+ *  the $mdColors.palette property lists the current selected palette
+ *  the $mdColors.colors array lists all the available colors with hex encoded color values for the current selected palette.
+ *   If you want to use any color for anything, you can query this array. The colors are available in hex notation.
+ *
+ **/
+
+
+    angular
+        .module('material.core.colors',['material.core.theming'])
+        .service('$mdColors', mdColors)
+        .config(configColors)
+        .run(loadDefaults);
+
+    /* @ngInject */
+    function configColors($mdThemingProvider) {
+        // fetch the colors out of the theming provider
+        Object.keys($mdThemingProvider._PALETTES).forEach(parsePalette);
+
+        // I need access to the default 'default', hardcoded for now as I miss access to it.
+        defaultPalette = $mdThemingProvider._THEMES['default'].colors.primary.name;
+        return;
+
+        // clone the palette colors to the colorStore var
+        function parsePalette(paletteName) {
+            var palette   = $mdThemingProvider._PALETTES[paletteName];
+            var hueColors = $mdThemingProvider._THEMES['default'].colors['primary'].hues;
+            var colors    = [];
+            colorStore[paletteName] = colors;
+            Object.keys(palette).forEach(copyColors);
+            Object.keys(hueColors).forEach(addHue);
+            return ;
+
+            function addHue(hueName) {
+                colors.push({
+                    color: hueName,
+                    value: palette[hueColors[hueName]]
+                });
+            }
+
+            function copyColors(colorName) {
```

Comment:
Shouldn't "colorName" be "paletteName"? This method gets called on a hash of palettes.